### PR TITLE
RDoc-2304 [Node.js] Client API > Session > How to > Perform operations lazily 

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issue:**
https://issues.hibernatingrhinos.com/issue/RDoc-2304/Nodejs-Client-API-Session-How-to-Perform-operations-laziliy

@ml054
For Node.js - the files to be reviewed are:

Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown
Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/lazy.js

Note: **5.3** files were only added due to links that have changed.

### CODE REVIEW

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+* [Getting compare-exchange](../../../client-api/session/cluster-transaction/compare-exchange#get-compare-exchange) values can be executed __lazily__.
+
+{CODE:nodejs lazy_compareExchange@ClientApi\Session\HowTo\lazy.js /}
+{NOTE/}
+
+{PANEL/}
+
+{PANEL: Execute all pending lazy operations}
+
+[//]: # (* To execute **all** pending lazy operations at once, use the `ExecuteAllPendingLazyOperations` method.  )
```
Comment (ml054) [inline Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown:99]: ExecuteAllPendingLazyOperations -> executeAllPendingLazyOperations

File: Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/lazy.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        //endregion
+    }
+    {
+        //region lazy_conditionalLoad
+        // Create document and get its change-vector:
+        {
+            const session1 = documentStore.openSession();
+
+            const employee = new Employee();
+            session1.store(employee, "employees/1-A");
```
Comment (ml054) [inline Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/lazy.js:59]: await ?

File: Documentation/5.3/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* [Getting compare-exchange](../../../client-api/session/cluster-transaction/compare-exchange#get-compare-exchange) values can be executed __lazily__.
+
+{CODE:nodejs lazy_compareExchange@client-api\Session\HowTo\lazy.js /}
+{NOTE/}
+
+{PANEL/}
+
+{PANEL: Execute all pending lazy operations}
+
+[//]: # (* To execute **all** pending lazy operations at once, use the `ExecuteAllPendingLazyOperations` method.  )
```
Comment (ml054) [inline Documentation/5.3/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown:99]: same here

File: Documentation/5.3/Samples/nodejs/client-api/Session/HowTo/lazy.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        //endregion
+    }
+    {
+        //region lazy_conditionalLoad
+        // Create document and get its change-vector:
+        {
+            const session1 = documentStore.openSession();
+
+            const employee = new Employee();
+            session1.store(employee, "employees/1-A");
```
Comment (ml054) [inline Documentation/5.3/Samples/nodejs/client-api/Session/HowTo/lazy.js:59]: same

File: Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown (hunk 1 above)
Comment (Danielle9897) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown:99]: Oh - this line is a comment that I forgot to remove
The next line has it correctly
Will remove this comment

Comment (ml054) [reply Documentation/5.2/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown:99]: parser in my eyes doesn't recognize comments :D

File: Documentation/5.3/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown (hunk 3 above)
Comment (Danielle9897) [reply Documentation/5.3/Raven.Documentation.Pages/client-api/session/how-to/perform-operations-lazily.js.markdown:99]: comment removed..

File: Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/lazy.js (hunk 2 above)
Comment (Danielle9897) [reply Documentation/5.2/Samples/nodejs/ClientApi/Session/HowTo/lazy.js:59]: added await

File: Documentation/5.3/Samples/nodejs/client-api/Session/HowTo/lazy.js (hunk 4 above)
Comment (Danielle9897) [reply Documentation/5.3/Samples/nodejs/client-api/Session/HowTo/lazy.js:59]: added await

**Review by ml054 — APPROVED** [review]

**Pull request merged**